src/mount/mastercomm.cc

```diff
@@ -476,261 +488,471 @@ void fs_register_config() {
-			fprintf(stderr,"can't set TCP_NODELAY\n");
+
+	if (!clientHasSessionFlags) { infoToPrint << "-"; }
+
+	if (!gInitParams.meta) {
+#ifndef _WIN32
+		struct passwd userEntry, *userInfo;
+		struct group groupEntry, *groupInfo;
+		std::vector<char> userGroupBuffer(kUserGroupBufferDefaultSize);
+
+		infoToPrint << " ; root mapped to ";
+		getpwuid_r(rootuid, &userEntry, userGroupBuffer.data(), kUserGroupBufferDefaultSize,
+		           &userInfo);
+		if (userInfo) {
+			infoToPrint << userInfo->pw_name << ":";
 		} else {
-			safs_pretty_syslog(LOG_WARNING,"can't set TCP_NODELAY");
+			infoToPrint << rootuid << ":";
 		}
-	}
-	if (srcip>0) {
-		if (tcpnumbind(fd,srcip,0)<0) {
-			if (verbose) {
-				fprintf(stderr,"can't bind socket to given ip (\"%s\")\n",srcstrip);
+
+		getgrgid_r(rootgid, &groupEntry, userGroupBuffer.data(), kUserGroupBufferDefaultSize,
+		           &groupInfo);
+		if (groupInfo) {
+			infoToPrint << groupInfo->gr_name;
+		} else {
+			infoToPrint << rootgid;
+		}
+
+		if (sesflags & SESFLAG_MAPALL) {
+			infoToPrint << " ; users mapped to ";
+			userInfo = getpwuid(mapalluid);
+			if (userInfo) {
+				infoToPrint << userInfo->pw_name << ":";
 			} else {
-				safs_pretty_syslog(LOG_WARNING,"can't bind socket to given ip (\"%s\")",srcstrip);
+				infoToPrint << mapalluid << ":";
 			}
-			tcpclose(fd);
-			fd=-1;
-			free(regbuff);
-			return -1;
+			groupInfo = getgrgid(mapallgid);
+			if (groupInfo) {
+				infoToPrint << groupInfo->gr_name;
+			} else {
+				infoToPrint << mapallgid;
+			}
+		}
+#else
+		infoToPrint << " ; root mapped to " << rootuid << ":" << rootgid;
+		// The Windows specific code makes consistent the use of client side
+		// uid/gid mappings and master side uid/gid mappings.
+		if (mapalluid != DEFAULT_UID_GID_MAPPING || mapallgid != DEFAULT_UID_GID_MAPPING) {
+			if (*mountingUid != USE_LOCAL_ID && *mountingGid != USE_LOCAL_ID) {
+				infoToPrint << " ; master server overwrote users mapping";
+			}
+			*mountingUid = mapalluid;
+			*mountingGid = mapallgid;
+		}
+
+		// At last, the final mapping is shown in the connection parameters
+		// line.
+		if (*mountingUid == USE_LOCAL_ID && *mountingGid == USE_LOCAL_ID) {
+			infoToPrint << " ; users mapped to local IDs";
+		} else {
+			infoToPrint << " ; users mapped to " << *mountingUid << ":" << *mountingGid;
+		}
+#endif
+	} else {
+		// meta
+		if (sesflags & SESFLAG_NONROOTMETA) {
+			nonRootAllowedToUseMeta() = true;
+		} else {
+			nonRootAllowedToUseMeta() = false;
 		}
 	}
-	if (tcpnumconnect(fd,masterip,masterport)<0) {
+	fprintf(stderr, "%s\n", infoToPrint.str().c_str());
+	safs_pretty_syslog(LOG_INFO, "%s", infoToPrint.str().c_str());
+}
+
+int fs_open_master_connection(bool verbose) {
+	int socket = tcpsocket();
+
+	if (socket < 0) { return -1; }
+
+	if (tcpnodelay(socket) < 0) {
 		if (verbose) {
-			fprintf(stderr,"can't connect to sfsmaster (\"%s\":\"%" PRIu16 "\")\n",masterstrip,masterport);
+			fprintf(stderr, "can't set TCP_NODELAY\n");
```

Comment:
Consider using the C++ 23 more modern `std::println()` function and the new logging format for all the modified lines.

```suggestion
			std::println(stderr, "can't set TCP_NODELAY");
```

Reply:
Considered. Maybe on another PR when we set a convention for this exact printing substitution.